pkg/operator/starter_controller.go

```diff
@@ -67,10 +67,13 @@ func (c *Controller) sync(ctx context.Context, syncCtx factory.SyncContext) erro
 		return nil
 	}
 
-	_, err = c.operatorControllerConfig.Precondition()
```

Comment:
I thought the idea earlier was - if `PreCondition` returns an error that means, preconditions has not been met and hence other controllers can't be started. What changed?

Reply:
That is still the case, the error is returned when the precondition is not valid within `setDisabledCondition` and is returned if it was valid but an error happened.
After taking another look on how that was managed in the legacy `csi-driver-manila-operator`, I noticed the `setDisabledCondition` is [only set when check manila shares wasn't ok](https://github.com/openshift/csi-driver-manila-operator/blob/master/pkg/controllers/manila/manila.go#L102-L114). I updated the commit to address that.

Reply:
Can we add some tests for this?